CRM/Autologcleanup/Job/LogCleanup.php

```diff
@@ -0,0 +1,105 @@
+<?php
+
+class CRM_Autologcleanup_Job_LogCleanup {
+
+  /**
+   * @var string
+   */
+  private $maxAge;
+
+  /**
+   * Constructs a new LogCleanup object.
+   *
+   * @param string $maxAge
+   *  Maximum log age.
+   */
+  public function __construct(string $maxAge) {
+    $this->maxAge = $maxAge;
+  }
+
+  /**
+   * Starts the scheduled job for clearing up
+   * stale log records.
+   *
+   * @return True
+   *
+   * @throws \CRM_Core_Exception
+   */
+  public function run() {
+    $this->pruneLogTables();
+
+    return TRUE;
+  }
+
+  /**
+   * Tranforms the log age to date ago.
+   *
+   * @param string $age
+   *  The age of the log to be cleared.
+   *
+   * @throws \CRM_Core_Exception;
+   */
+  private function transformLogAge(string $age) {
+    $timestamp = strtotime("today - $age");
+    if (!$timestamp) {
+      throw new \CRM_Core_Exception("${age} is an invalid log max age");
+    }
+
+    return date('Y-m-d', $timestamp);
+  }
+
+  /**
+   * Prunes stale records from log tables.
+   *
+   * Records that have stayed loger than the max age
+   * are considered stale.
+   */
+  private function pruneLogTables() {
+    $ageAgo = $this->transformLogAge($this->maxAge);
+    $logTables = $this->getLogTables();
+
+    foreach ($logTables as $logTable) {
+      $query = CRM_Utils_SQL_Delete::from($logTable)
+        ->where('log_date < @date', ['date' => $ageAgo]);
+      $query->execute();
+    }
+  }
+
+  /**
+   * Gets all the log tables
+   *
+   * @return array
+   *   Returns an array of log tables
+   */
+  private function getLogTables() {
+    $logTables = [];
+    $loggingSchema = new \CRM_Logging_Schema();
+    $nonStandardTableNameString = $loggingSchema->getNonStandardTableNameFilterString();
+    if (defined('CIVICRM_LOGGING_DSN')) {
+      $dsn = CRM_Utils_SQL::autoSwitchDSN(CIVICRM_LOGGING_DSN);
+      $dsn = DB::parseDSN($dsn);
+      $this->useDBPrefix = (CIVICRM_LOGGING_DSN != CIVICRM_DSN);
+    }
+    else {
+      $dsn = CRM_Utils_SQL::autoSwitchDSN(CIVICRM_DSN);
+      $dsn = DB::parseDSN($dsn);
+      $this->useDBPrefix = FALSE;
+    }
+    $this->db = $dsn['database'];
+
+    $dao = CRM_Core_DAO::executeQuery("
+      SELECT TABLE_NAME
+      FROM   INFORMATION_SCHEMA.TABLES
+      WHERE  TABLE_SCHEMA = '{$this->db}'
+      AND    TABLE_TYPE = 'BASE TABLE'
+      AND    (TABLE_NAME LIKE 'log_civicrm_%' $nonStandardTableNameString )
```

Comment:
We don't need the $nonStandardTableNameString part, because there are many extension specific tables that do not start with `civicrm_` (like Membershipextras), and the last thing we want is to delete their data.

Reply:
It's actually for getting the log tables that do not start with the conventional `log_civicrm`. we can see the comment here https://github.com/compucorp/civicrm-core/blob/9c3e897c6bf3593f26b3210cd2bcb90670ed5b88/CRM/Logging/Schema.php#L893-L902.

I will remove it for now, and add it back if we agree it is safe.
cc @omarabuhussein

Reply:
@olayiwola-compucorp I saw that, and it comes from this ticket:  https://issues.civicrm.org/jira/browse/CRM-18180
which says that logs tables can be altered using hooks, which I think they mean this hook: 
https://docs.civicrm.org/dev/en/latest/hooks/hook_civicrm_alterLogTables/

which is not a hook that is common to be used by extensions, I also checked our Compuclient and only one extension uses it to prevent Civi from creating log tables for that extension.

Reply:
Ah nevermind, I just noticed that it prefixes these tables with `log_`